src/plugins/GenerateJob/templates/plotly_backend.py

```diff
@@ -0,0 +1,742 @@
+        props['tickformat'] = list(formatter.seq)
+    elif not any(label.get_visible() for label in axis.get_ticklabels()):
+        props['tickformat'] = ""
+    else:
+        props['tickformat'] = None
+
+    # Get axis scale
+    props['scale'] = axis.get_scale()
+
+    # Get major tick label size (assumes that's all we really care about!)
+    labels = axis.get_ticklabels()
+    if labels:
+        props['fontsize'] = labels[0].get_fontsize()
+    else:
+        props['fontsize'] = None
+
+    # Get associated grid
+    props['grid'] = mplexporter.utils.get_grid_style(axis)
+
+    # get axis visibility
+    props['visible'] = axis.get_visible()
+
+    return props
+
+
+def get_symbol_3d(marker_symbol):
+    """convert mpl marker symbols into supported plotly 3d symbols"""
+    symbol = mpltools.convert_symbol(marker_symbol)
+    if symbol not in PLOTLY_3D_MARKER_SYMBOLS:
+        return 'circle'
+
+
+def convert_z_domain(mpl_plot_bounds, mpl_max_z_bounds):
+    """Get domain bounds for a 3d-ZAxis matplotlib"""
+    mpl_z_dom =  [mpl_plot_bounds[2], mpl_plot_bounds[2] + mpl_plot_bounds[3]]
+    plotting_depth = mpl_max_z_bounds[1] - mpl_max_z_bounds[0]
+    z0 = (mpl_z_dom[0] - mpl_max_z_bounds[0]) / plotting_depth
+    z1 = (mpl_z_dom[1] - mpl_max_z_bounds[0]) / plotting_depth
+    return [z0, z1]
+
+
+def prep_xyz_axis(ax, props, x_bounds, y_bounds, z_bounds):
+    """Crawl properties for a  matplotlib Axes3D"""
+    xaxis = dict(
+        type=props['axes'][0]['scale'],
+        range=list(props['xlim']),
+        domain=mpltools.convert_x_domain(props['bounds'], x_bounds),
+        side=props['axes'][0]['position'],
+        tickfont=dict(size=props['axes'][0]['fontsize'])
+    )
+    xaxis.update(mpltools.prep_ticks(ax, 0, 'x', props))
+
+    yaxis = dict(
+        type=props["axes"][1]["scale"],
+        range=list(props["ylim"]),
+        showgrid=props["axes"][1]["grid"]["gridOn"],
+        domain=mpltools.convert_y_domain(props["bounds"], y_bounds),
+        side=props["axes"][1]["position"],
+        tickfont=dict(size=props["axes"][1]["fontsize"]),
+    )
+
+    yaxis.update(mpltools.prep_ticks(ax, 1, "y", props))
+
+    zaxis = dict(
+        type=props['axes'][2]['scale'],
+        range=list(props['zlim']),
+        showgrid=props['axes'][1]['grid']['gridOn'],
+        side=props['axes'][2]['position'],
+        tickfont=dict(size=props['axes'][2]['fontsize'])
+    )
+
+    zaxis.update(mpltools.prep_ticks(ax, 2, "z", props))
+
+    return xaxis, yaxis, zaxis
+
+
+def mpl_to_plotly(fig):
+    """Convert matplotlib figure to a plotly figure
+
+    Parameters
+    ----------
+    fig : matplotlib.pyplot.Figure
+        The matplotlib figure
+
+    Returns
+    -------
+    plotly.graph_objects.Figure
+        The converted plotly Figure
+    """
+    renderer = DeepforgePlotlyRenderer()
+    exporter = mplexporter.Exporter(renderer)
+    exporter.run(fig)
+    renderer.crawl_3d_labels(fig)
```

Comment:
`renderer.resize()` method removes width, height and other properties for a plotly figure, which is fine. But, this is already done in PlotyGraphWidget. Can be done here as well.